examples/typescript/example-audiocallback.ts

```diff
@@ -0,0 +1,218 @@
+            }
+        }
+    }
+
+    // BUFFER_OUT for the buses
+    if (event.command === AudioCallbackCommands.BUFFER_OUT) {
+        const { inputChannels, samplesPerFrame, outputChannels } = event.data;
+        for (const [ch, inputChannel] of inputChannels.entries()) {
+            // Don't copy audio signal if it should be muted
+            if (MUTE_BUS_CHANNELS.has(ch)) {
+                continue;
+            }
+            for (let i = 0; i < samplesPerFrame; i++) {
+                outputChannels[ch][i] = inputChannel[i];
+            }
+        }
+    }
+};
+
+// Swaps the channels of the first 2 output devices
+const swapTwoOutputDevices = (event: types.AudioCallbackEvent): void => {
+    if (event.command === AudioCallbackCommands.BUFFER_OUT) {
+        const { inputChannels, samplesPerFrame, outputChannels } = event.data;
+        // Takes the first 8 input channels, and copies the signal to the output channels 8-15
+        for (const [ch, inputChannel] of inputChannels.slice(0, 8).entries()) {
+            const outputChannel = outputChannels[ch + 8];
+            for (let i = 0; i < samplesPerFrame; i++) {
+                outputChannel[i] = inputChannel[i];
+            }
+        }
+        // Takes the input channels 8-15, and copies them to the output channels 0-7
+        for (const [ch, inputChannel] of inputChannels.slice(8, 16).entries()) {
+            const outputChannel = outputChannels[ch];
+            for (let i = 0; i < samplesPerFrame; i++) {
+                outputChannel[i] = inputChannel[i];
+            }
+        }
+    }
+};
+
+// Record audio signal of the first bus
+const recordedSamples: Array<number[]> = Array.from({ length: 8 }, () => []);
+
+const recordOutputChannel = (event: types.AudioCallbackEvent) => {
+    if (event.command === AudioCallbackCommands.STARTING) {
+        console.log("Recording started");
+    }
+    if (event.command === AudioCallbackCommands.BUFFER_OUT) {
+        const { inputChannels, samplesPerFrame, outputChannels } = event.data;
+        // Store the samples of the first 8 channels, and passthrough the signal
+        for (const [ch, inputChannel] of inputChannels.slice(0, 8).entries()) {
+            for (let i = 0; i < samplesPerFrame; i++) {
+                recordedSamples[ch].push(inputChannel[i]);
+                outputChannels[ch][i] = inputChannel[i];
+            }
+        }
+        // Directly passthrough audio signal
+        for (const [ch, inputChannel] of inputChannels.slice(8).entries()) {
+            for (let i = 0; i < samplesPerFrame; i++) {
+                outputChannels[ch + 8][i] = inputChannel[i];
+            }
+        }
+    }
+    if (event.command === AudioCallbackCommands.ENDING) {
+        console.log("Recording finished. Total samples:", recordedSamples[0].length);
+    }
+};
+
+// Output recorded audio signal to first bus
+let playbackPosition = 0;
+const playAudioFile = (event: types.AudioCallbackEvent) => {

```

Comment:
This should be changed to `playRecordedAudio`, originally I had it play an audio file and forgot to update the naming of it.